src/Meshes/BoxMesh.jl

```diff
@@ -0,0 +1,85 @@
+"""
+    rectangular_mesh(x1c, x2c, per = (false, false))
+
+This function builds a 2D rectangular mesh with points located at x1c and x2c in x1 and x2 directions respectively.
+"""
+function rectangular_mesh(x1c, x2c, per = (false, false))
```

Comment:
We have already a rectangular mesh. How is this different? And they are located in Meshes/Meshes.jl. Probably no need for a separate file

Reply:
The rectangle_mesh function generates output in the Mesh2D format (conformal, unstructured). The advantage is that all 2D unstructured meshes (ex: cube surface mesh) can be handled by the same code. 
The interface `rectangular_mesh(x1c, x2c, per = (false, false))`, covers all 4 sided polygons (assuming warping function is applied to the rectangular domain). Equispaced, stretched or even manually spaced point locations can all be covered as wrappers around this function, where the wrappers feed in the respective x1c and x2c.

Reply:
If it's for comparison purposes, then it would probably make sense to (a) have a method to convert the structured to the unstructured, and (b) a unit test that tests them